src/Compilers/Core/Portable/Diagnostic/DiagnosticFormatter.cs

```diff
@@ -101,6 +109,34 @@ internal string GetMessagePrefix(Diagnostic diagnostic, CultureInfo culture)
                 diagnostic.Id);
         }
 
-        internal static readonly DiagnosticFormatter Instance = new DiagnosticFormatter();
+        internal string Format(Diagnostic diagnostic, IFormatProvider formatter, AnalyzerManager analyzerManager)
+        {
+#pragma warning disable RS0013 // we do need descriptor instance here
+            if (!_displayAnalyzer || analyzerManager == null || diagnostic.Descriptor == null)
+            {
+                return Format(diagnostic, formatter);
+            }
+
+            var analyzerIdentity = GetAnalyzerIdentity(analyzerManager.GetDiagnosticAnalyzer(diagnostic.Descriptor));
+            if (analyzerIdentity == null)
+            {
+                return Format(diagnostic, formatter);
+            }
+
+            return Format(diagnostic, formatter) + ": " + analyzerIdentity;
+#pragma warning restore RS0013 
+        }
+
+        protected virtual string GetAnalyzerIdentity(DiagnosticAnalyzer analyzer)
+        {
+            if (analyzer == null)
+            {
+                return null;
+            }
+
+            return analyzer.GetType().ToString();
+        }
+
+        internal static readonly DiagnosticFormatter Instance = new DiagnosticFormatter(displayAnalyzer: false);
```

Comment:
It seems weird that CSharpDiagnosticFormatter.Instance is public but this one isn't. Not related to your change, but still it seems that even this should be public.